.github/workflows/functional-test.yaml

```diff
@@ -100,12 +100,27 @@ jobs:
           go install github.com/fullstorydev/grpcurl/cmd/grpcurl@latest
           grpcurl -version
 
-      - name: run functional test
-        run:  bash functional_test.sh ${{ github.sha }}-server-dev
+      - name: Run functional test queries
+        run: |
+          bash query_functional_test.sh ${{ github.sha }}-server-dev
+
+      # todo: use job-specific archive names to prevent parallel jobs from overwriting each other's results?
+      # Does it matter, given the results will be analyzed instantly after this?
+      - name: Archive street functional test results
```

Comment:
This is a coool approach. I like how clean it is (i.e. doesn't pollute the repo with test data), but how easy is it to extract/explore the output after the fact? is there a command or instructions to add to grab the archived data?

Reply:
Good question! From my [reading](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts) on github actions archives, it seems like you can use these results in later workflow runs by just "[downloading](https://docs.github.com/en/actions/guides/storing-workflow-data-as-artifacts#downloading-or-deleting-artifacts)" the archive by name (so, this one would be called `functional-test-street-results` and you can import that into a later-running GH actions workflow with that name). 

I have a comment here about the possibility of making these artifact names unique per-run, because there might (?) be race conditions possible if 2 CI runs kick off around the same time. But I don't think that's truly a concern for the main functional test workflow here, because in the next PR we're adding a step that runs right and pulls these test results in to analyze them (ie, race conditions are highly unlikely, I'd guess).

This is also the reason why I think the golden set results will get overwritten each time the manual GH action to generate it is run, because I use a static name. There's an argument for unique names there as well, so happy to hear opinions

Reply:
yeah, it doesn't seem like we'll be generating 2 golden sets at the same time, but good to point out.

> This is also the reason why I think the golden set results will get overwritten each time the manual GH action to generate it is run, because I use a static name. There's an argument for unique names there as well, so happy to hear opinions

here's where I think unique names might be worth it. We don't want to accidentally overwrite (unless it's easy to recover)

Reply:
I was going to come in and suggest using unique names, but I followed some of the links you shared and I think that the artifacts are going to be (effectively) namespaced by workflow run anyway. I.e. there is _not_ a single location that will be overwritten by each workflow. Worth experimenting with though, sounds like we're taking different interpretations away from the docs

Reply:
Hmm re-read the docs again and I think you're right @danielhfrank . That means that we should still be fine using GH artifacts for the main functional test workflow, but will need a different approach for the golden set generation action, if we want to reference output of previously-run workflows.

I think I'll try uploading/downloading golden set results from GCS, that seems like the next best option

Reply:
Created a [new bucket](https://console.cloud.google.com/storage/browser/graphhopper_test_artifacts;tab=objects?project=model-159019&prefix=&forceOnObjectsSortingFiltering=false) for these golden response set artifacts, and just pushed an update to that workflow to upload these results there instead of storing them as artifacts. We unfortunately can't test this until it's merged, but it follows similar things we've done in other workflows we have currently in use.

My plan for pulling the latest golden result sets from GCS will be to do something like [this](https://stackoverflow.com/a/54218908) - basically just pulling the last-added file in each folder in that bucket to do analysis against